media/js/firefox/australis/growth-firstrun-test1.js

```diff
@@ -0,0 +1,217 @@
+        // initialize vars
+        $body = $('body');
+        $ghosteryIframe = $('iframe.addon');
+
+        fxaClientID = window.FXA_RELIER.clientId;
+
+        // determine whether or not to show doorhanger (50% of users)
+        showDoorhanger = Math.random() >= 0.5;
+
+        // GA
+        gaVariation = (showDoorhanger) ? 'a' : 'b';
+
+        window.gaTrack(['_setCustomVar', 7, 'first run tests', 'variation 2' + gaVariation, 2]);
+        window.gaTrack(['_trackEvent','/firstrun/ A/B Test', 'page load', 'variation 2' + gaVariation]);
+
+        enableIframe = function() {
+            $ghosteryIframe.attr('src', $ghosteryIframe.attr('data-src'));
+        };
+
+        showFxA = function() {
+            $fxaOverlay.addClass('visible');
+
+            // display heading with timeout (so transition happens)
+            setTimeout(function() {
+                $fxaValueProp.addClass('visible');
+            }, 500);
+
+            // display sub-CTAs 40s later
+            // temporary until FxA client supports broadcast of 'Signup' click
+            setTimeout(function() {
+                $fxaDelayedCTAs.addClass('visible');
+            }, 40000);
+        };
+
+        initFxA = function() {
+            // set up FxA client
+            fxaClient = new window.FxaRelierClient(fxaClientID, {
+                contentHost: window.FXA_RELIER.contentHost,
+                oauthHost: window.FXA_RELIER.contentOauth
+            });
+
+            // conjure FxA lightbox
+            fxaClient.auth.signUp({
+                redirectUri: 'http://localhost:8000/',
+                scope: 'profile',
+                state: '123456', // TODO: create real CSRF token
+                ui: 'lightbox',
+                background: '#fff',
+                zIndex: 99,
+                keepOverlay: true // customized build of fxa-relier-client.js to support this option
+            }).then(function(result) {
+                // make sure CSRF/state matches
+                if (result.state === '123456') {
+                    // change header
+                    $('#fxa-value-prop-intro').fadeOut('fast', function() {
+                        $('#fxa-value-prop-verified').fadeIn('fast');
+                    });
+
+                    // wipe out iframe and reposition delayed CTAs
+                    $fxaIframe.slideUp('normal', function() {
+                        $fxaIframe.remove();
+                    });
+
+                    $fxaDelayedCTAs.addClass('visible noiframe');
+                }
+            }, function(err) {
+                // on error, get rid of overlay and fall back to default content
+                $fxaOverlay.remove();
+                showDefaultContent();
+            });
+        };
+
+        // listen for postMessage from iframes
+        window.addEventListener('message', function(event) {
```

Comment:
Should we be verifying the origin of these cross iframe messages? https://developer.mozilla.org/en-US/docs/Web/API/Window.postMessage#Security_concerns